pymedphys/_experimental/pseudonymisation/__init__.py

```diff
@@ -1,8 +1,108 @@
-from pymedphys._dicom.anonymise import anonymise_cli
+# Copyright (C) 2020 Stuart Swerdloff, Simon Biggs
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import json
+import logging
+from os.path import abspath, basename, dirname, isdir, isfile
+from os.path import join as pjoin
+
+from pymedphys._dicom.anonymise import (
+    anonymise_directory,
+    anonymise_file,
+    get_default_identifying_keywords,
+)
+
+from . import strategy
+
+HERE = dirname(abspath(__file__))
+
+IDENTIFYING_UIDS_FILEPATH = pjoin(HERE, "identifying_uids.json")
+
+
+@functools.lru_cache()
+def _get_default_identifying_uids():
+    with open(IDENTIFYING_UIDS_FILEPATH) as infile:
+        IDENTIFYING_UIDS = json.load(infile)
+    return tuple(IDENTIFYING_UIDS)
+
+
+def get_default_identifying_uids():
+    return list(_get_default_identifying_uids())
+
+
+@functools.lru_cache()
+def _get_default_pseudonymisation_keywords():
+    anon_keyword_set = set(get_default_identifying_keywords())
+    psuedo_uid_set = set(get_default_identifying_uids())
+    return tuple(anon_keyword_set.union(psuedo_uid_set))
+
+
+def get_default_pseudonymisation_keywords():
+    return list(_get_default_pseudonymisation_keywords())
 
 
 def anonymise_with_pseudo_cli(args):
+
+    if args.delete_unknown_tags:
+        handle_unknown_tags = True
+    elif args.ignore_unknown_tags:
+        handle_unknown_tags = False
+    else:
+        handle_unknown_tags = None
+
+    if not args.keywords_to_leave_unchanged:
+        keywords_to_leave_unchanged = ()
+    else:
+        keywords_to_leave_unchanged = args.keywords_to_leave_unchanged
+
+    replacement_strategy = None
     if args.pseudo:
-        print("Was run with pseudo!")
+        logging.info("Was run with pseudo!")
+        identifying_keywords_for_pseudo = get_default_pseudonymisation_keywords()
```

Comment:
new list of identifying keywords, which is basically the uids plus the identifying keywords already known to the anonymise module (the union of them, to avoid a problem if someone adds the uids back to the anonymise module).